File: sentimental-credit/credit.py
# TODO

def main():
    n = input("Number: ")
    numDigits = len(n)
    number = int(n)

    odd = 0
    even = 0
    digit = 0

    for i in range(numDigits):
        if i % 2 != 0:
            digit = (number % 10) * 2
            if digit > 9:
                odd += digit - 9
            else:
                odd += digit
        else:
            even += (number % 10)

        number //= 10

    sum = odd + even

    if sum % 10 == 0:
        firstTwo = int(n[0] + n[1])
        first = int(n[0])
        if (firstTwo >= 51) and (firstTwo <= 55):
            print("MASTERCARD")
        elif (first == 4) and (numDigits == 13 or numDigits == 16):
            print("VISA")
        elif (firstTwo == 34 or firstTwo == 37) and (numDigits == 15):
            print("AMEX")
        else:
            print("INVALID")
    else:
        print("INVALID")


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
int main(void)
{
    long n = get_long("Number: ");
    long temp = n;
    int odd = 0;
    int even = 0;
    int digit;
    int numDigits = 0;
    long t = n;
    while (t > 0)
    {
        t /= 10;
        numDigits++;
    }
    int firstNumber;
    int secondNumber;
    for (int i = 0; i < numDigits; i++)
    {
        if (i == (numDigits - 1))
        {
            firstNumber = (temp % 10);
        }
        if (i == (numDigits - 2))
        {
            secondNumber = (temp % 10);
        }

        if (i % 2 != 0)
        {
            digit = (temp % 10) * 2;
            if (digit > 9)
            {
                odd += digit - 9;
            }
            else
            {
                odd += digit;
            }
        }
        else
        {
            even += (temp % 10);
        }

        temp /= 10;
    }
    int sum = odd + even;
    if (sum % 10 == 0)
    {
        if (firstNumber == 5 && (secondNumber >= 1 && secondNumber <= 5))
        {
            printf("MASTERCARD\n");
        }
        else if (firstNumber == 4 && (numDigits == 13 || numDigits == 16))
        {
            printf("VISA\n");
        }
        else if (firstNumber == 3 && (secondNumber == 4 || secondNumber == 7))
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}